src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -422,7 +422,7 @@ internal unsafe void Start()
                         // The base FileSystemWatcher does a match check against the relative path before combining with
                         // the root dir; however, null is special cased to signify the root dir, so check if we should use that.
                         ReadOnlySpan<char> relativePath = ReadOnlySpan<char>.Empty;
-                        if (!path.Equals(_fullDirectory, StringComparison.OrdinalIgnoreCase))
+                        if (path.Length > _fullDirectory.Length && path.StartsWith(_fullDirectory, StringComparison.OrdinalIgnoreCase))
```

Comment:
Does `_fullDirectory` have a trailing slash? If not this won't be right.  `/abcdef` is not within `/abc`.

Reply:
> Does _fullDirectory have a trailing slash?

Yes:
https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs#L290-L294